pkg/controllers/testing/table.go

```diff
@@ -56,21 +57,27 @@ type Testcase struct {
 	// ExpectCreates holds the ordered list of objects expected to be created during reconciliation
 	ExpectCreates []runtime.Object
 
-	// ExpectStatusUpdates holds the ordered list of objects whose status is updated during reconciliation.
+	// ExpectStatusUpdates holds the ordered list of objects whose status is updated during reconciliation
 	ExpectStatusUpdates []runtime.Object
 
 	// ShouldErr is true if and only if reconciliation is expected to return an error
 	ShouldErr bool
 
-	// ExpectErrMessage is the error message expected if ShouldErr is true
-	ExpectErrMessage string
+	// ShouldRequeue is true if and only if reconciliation is expected to return a Result with Requeue set to true
+	ShouldRequeue bool
+
+	// Verify provides the reconciliation Result and error for custom assertions
+	Verify VerifyFunc
 
 	// Hooks, if provided, wrap the corresponding client method. They are typically used to inject errors. The object
 	// will be recorded as created if and only if the hook calls the wrapped method.
 	CreateHook CreateHook
 	GetHook    GetHook
```

Comment:
for another PR: it would be helpful to document `{Create,Get}Hook`, I currently do not understand what purpose they serve when looking at their usage in the tests

Reply:
Tried to improve that and added an example. See also the godoc for the `CreateHook` and `GetHook` types.